src/main/fc/fc_msp.c

```diff
@@ -1949,6 +1955,30 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
     UNUSED(dataSize); // maybe unused due to compiler options
 
     switch (cmdMSP) {
+    case MSP_SET_STREAM:
+        {
+            uint8_t flags = sbufReadU8(src);
+            if (flags & MSP_STREAM_FLAG_BEGIN) {
+
+                uint8_t entryCount = sbufReadU8(src);
+
+                for (int i = 0; i < entryCount; i++) {
+
+                    uint8_t flags = sbufReadU8(src);
+                    uint16_t id = sbufReadU16(src);
+                    uint8_t hz = sbufReadU8(src);
+
+                    extern mspPort_t *mspPortBeingProcessed;
```

Comment:
yes, this is proof of concept code at this point for discussion purposes, further cleanup will be done.  all your points are noted/would be addressed.